### ISSUE TITLE
mongodb extension timeout dont work

### ISSUE THREAD

**Issue by pinkladywithcoco** [mongodb/mongo-php-driver#271]
On the old client, you can set it like that timeout:

```
try {

    $mongo = new MongoClient("mongodb://domainsharded.com:27017,domainsharded2.com:27017",

                             array("connectTimeoutMS" => 1000)

                             );

} catch (MongoConnectionException $e) {

        echo "Connection fail";
}
```

This code will try to connect to the first mongos instance if that connection is not made during 1 second it will try to connect the second one. If both cant then it will say: Connection fail. I have tested it and it works.

But i want to use the new mongodb extension that isavailable from PECL to do the same. So i did:
```
try {

    $mongo = new MongoDB\Driver\Manager("mongodb://domainsharded.com:27017,domainsharded2.com:27017",

                             array("connectTimeoutMS" => 1000)

                             );

} catch (Exception $e) {

        echo "Connection fail";
}
```

This do not timeout during 1 second. It will also not try the second mongos instance if the first one fails. it does not throw anything. Please help me to make it work like the old client and show me how.

i'm using ubuntu 14.04. Have tested on php 5.6 and php 7 same thing happens.

**Comment by jmikola** [comment]
Cross-referencing with [PHPC-655](https://jira.mongodb.org/browse/PHPC-655) and discussion in https://github.com/mongodb/mongo-php-library/issues/145.

**Comment by jmikola** [comment]
[1.2.0alpha2](https://github.com/mongodb/mongo-php-driver/releases/tag/1.2.0alpha2) has been released, which includes significant changes to our connection handling and persistence. We've entirely dropped PHP's streams API and now rely on libmongoc for all socket communication, which I expect may resolve a number of outstanding connection handling and timeout issues.

@pinkladywithcoco: Please attempt reproducing your original problem with 1.2.0alpha2 and let us know if the issue persists. Thanks!

**Comment by jmikola** [comment]
Note: we've just released [1.1.9](https://github.com/mongodb/mongo-php-driver/releases/tag/1.1.9), which includes a back-ported fix (in libmongoc) for spurious timeouts ""No suitable servers found" exceptions when one or more replica sets nodes were inaccessible. The bug was specific to the PHP stream API integration, so it does not apply to version [1.2.0alpha2](https://github.com/mongodb/mongo-php-driver/releases/tag/1.2.0alpha2) and later.

Since this issue has been sitting for some time without confirmation, I'm going to close it out. Feel free to follow-up here or open a new issue as needed.
